eng/common/pipelines/templates/steps/verify-samples.yml

```diff
@@ -0,0 +1,19 @@
+parameters:
+  - name: ServiceDirectory
+    type: string
+    default: not-specified
+  - name: ScriptDirectory
+    type: string
+    default: eng/common/scripts
+  - name: Condition
+    type: boolean
+    default: succeeded()
+
+steps:
+  - pwsh: |
+      # If the last path segment is an absolute path it will be used entirely.
+      $root = [System.IO.Path]::Combine('$(Build.SourcesDireectory)', 'sdk', '${{ parameters.ServiceDirectory }}')
```

Comment:
Why not use Join-Path?

Reply:
That, and `Join-Path` doesn't handle other scenarios that `Path.Combine` does as well. Using `Path.Combine` over `Join-Path` solves many problems and regular PS scripters will use it for file system paths. At one point (though it seems fixed "now") it was little more than string concatenation. Seems with PowerShell (Core) they might've improved that since it had to normalize path separators for the host.